### PR TITLE
[Dialogs] Enable custom fonts on MDCAlertController

### PR DESCRIPTION
**Pull request by ianegordon**

### CODE REVIEW

**Comment by ianegordon** [comment]
screen shot 2018-03-14 at 16 34 23
screen shot 2018-03-14 at 16 35 00
screen shot 2018-03-14 at 16 33 00

File: components/Dialogs/src/MDCAlertController.m [hunk 1]
```diff
@@ -58,29 +60,10 @@ - (id)copyWithZone:(__unused NSZone *)zone {
...
-
-@property(nonatomic, strong) UIScrollView *contentScrollView;
-@property(nonatomic, strong) UIScrollView *actionsScrollView;
-
-@property(nonatomic, strong) UILabel *titleLabel;
-@property(nonatomic, strong) UILabel *messageLabel;
-
-@property(nonatomic, getter=isVerticalActionsLayout) BOOL verticalActionsLayout;
+@property(nonatomic, nullable, weak) MDCAlertControllerView *alertView;
+//@property(nonatomic, nonnull, strong) NSMutableArray<UIButton *> *actionButtons;
```
Comment (yarneo) [inline components/Dialogs/src/MDCAlertController.m]: please remove the comment

File: components/Dialogs/src/MDCAlertController.m [hunk 2]
```diff
@@ -256,47 +225,33 @@ - (void)actionButtonPressed:(id)sender {
 
 #pragma mark - UIViewController
 
+- (void)loadView {
+  self.view = [[MDCAlertControllerView alloc] initWithFrame:CGRectZero];
+  self.alertView = (MDCAlertControllerView *)self.view;
```
Comment (yarneo) [inline components/Dialogs/src/MDCAlertController.m:225]: everything other than instantiating `self.view` should probably be moved to `viewDidLoad`

File: components/Dialogs/src/MDCAlertController.m [hunk 3]
```diff
@@ -177,25 +162,14 @@ - (void)setMessage:(NSString *)message {
...
+        [self.alertView calculatePreferredContentSizeForBounds:CGRectInfinite.size];
 
-  self.preferredContentSize = [self calculatePreferredContentSizeForBounds:CGRectInfinite.size];
-
-  [self.view setNeedsLayout];
+    [self.alertView setNeedsLayout];
+  }
 }
 
 - (BOOL)mdc_adjustsFontForContentSizeCategory {
```
Comment (yarneo) [inline components/Dialogs/src/MDCAlertController.m:170]: `return self.alertView.mdc_adjustsFontForContentSizeCategory` we can remove the additional backing store BOOL in the top of the file.

File: components/Dialogs/src/MDCAlertController.m [hunk 4]
```diff
@@ -205,10 +179,7 @@ - (BOOL)mdc_adjustsFontForContentSizeCategory {
 - (void)mdc_setAdjustsFontForContentSizeCategory:(BOOL)adjusts {
   _mdc_adjustsFontForContentSizeCategory = adjusts;
```
Comment (yarneo) [inline components/Dialogs/src/MDCAlertController.m:175]: remove backing store

File: components/Dialogs/src/MDCAlertController.m [hunk 5]
```diff
@@ -205,10 +179,7 @@ - (BOOL)mdc_adjustsFontForContentSizeCategory {
...
 
-  // Update any action buttons to match the alert controller.
-  for (MDCButton *actionButton in self.actionButtons) {
-    actionButton.mdc_adjustsFontForContentSizeCategory = adjusts;
-  }
+  self.alertView.mdc_adjustsFontForContentSizeCategory = adjusts;
 
   if (_mdc_adjustsFontForContentSizeCategory) {
```
Comment (yarneo) [inline components/Dialogs/src/MDCAlertController.m:179]: can we move this to the view and just call its getter? if not, use `if (adjusts)` or alertView's bool.

File: components/Dialogs/src/MDCAlertController.h [hunk 6]
```diff
@@ -92,6 +92,7 @@
 
  Default value is NO.
  */
+// TODO : Deprecate in favor of MDCAlertControllerView
```
Comment (yarneo) [inline components/Dialogs/src/MDCAlertController.h]: can we add a `__deprecated_msg`? any reason why this is a TODO?

File: components/Dialogs/src/MDCAlertControllerView.h [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+@property(nonatomic, strong, nullable) UIColor *titleColor UI_APPEARANCE_SELECTOR;
+
+@property(nonatomic, strong, nullable) UIFont *messageFont UI_APPEARANCE_SELECTOR;
+@property(nonatomic, strong, nullable) UIColor *messageColor UI_APPEARANCE_SELECTOR;
+
+@property(nonatomic, strong, nullable) UIFont *buttonFont UI_APPEARANCE_SELECTOR;
+@property(nonatomic, strong, nullable) UIColor *buttonColor UI_APPEARANCE_SELECTOR;
+
+// TODO: Document how to theme buttons via UIAppearance proxy
+//??? @property(nonatomic, strong, nullable) UIFont *actionFont UI_APPEARANCE_SELECTOR;
```
Comment (yarneo) [inline components/Dialogs/src/MDCAlertControllerView.h]: do we want to leave these?

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 8]
```diff
@@ -0,0 +1,597 @@
...
+      self.messageLabel.font = [UIFont mdc_preferredFontForMaterialTextStyle:MDCFontTextStyleBody1];
+    } else {
+      self.messageLabel.font = [MDCTypography body1Font];
+    }
+    self.messageLabel.textColor = [UIColor colorWithWhite:0.0f alpha:MDCDialogMessageOpacity];
+    [self.contentScrollView addSubview:self.messageLabel];
+
+    _actionButtons = [[NSMutableArray alloc] init];
+
+//    CGSize idealSize = [self calculatePreferredContentSizeForBounds:CGRectInfinite.size];
```
Comment (yarneo) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m]: remove?

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 9]
```diff
@@ -0,0 +1,597 @@
...
+
+//    self.preferredContentSize = idealSize;
```
Comment (yarneo) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m]: remove?

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 10]
```diff
@@ -0,0 +1,597 @@
...
+}
+
+- (NSString *)title {
+  return self.titleLabel.text;
+}
+
+- (void)setTitle:(NSString *)title {
+  self.titleLabel.text = title;
+
+//  self.preferredContentSize = [self calculatePreferredContentSizeForBounds:CGRectInfinite.size];
```
Comment (yarneo) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m]: remove?

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 11]
```diff
@@ -0,0 +1,597 @@
...
+}
+
+- (NSString *)message {
+  return self.messageLabel.text;
+}
+
+- (void)setMessage:(NSString *)message {
+  self.messageLabel.text = message;
+
+//  self.preferredContentSize = [self calculatePreferredContentSizeForBounds:CGRectInfinite.size];
```
Comment (yarneo) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m]: remove?

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 12]
```diff
@@ -0,0 +1,597 @@
...
+
+  [self setNeedsLayout];
```
Comment (yarneo) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m]: do we need this here and in the VC? Can this be called directly?

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 13]
```diff
@@ -0,0 +1,597 @@
...
+- (NSString *)title {
+  return self.titleLabel.text;
+}
+
+- (void)setTitle:(NSString *)title {
+  self.titleLabel.text = title;
+
+//  self.preferredContentSize = [self calculatePreferredContentSizeForBounds:CGRectInfinite.size];
+
+  [self setNeedsLayout];
```
Comment (yarneo) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m]: do we need this here and in the VC? Can this be called directly?

**Review by yarneo — DISMISSED** [review]

File: components/Dialogs/src/MDCAlertController.m (hunk 3 above)
Comment (ianegordon) [reply components/Dialogs/src/MDCAlertController.m:170]: Unfortunately that won't work if alertView hasn't been instantiated yet.

File: components/Dialogs/src/MDCAlertController.m (hunk 2 above)
Comment (ianegordon) [reply components/Dialogs/src/MDCAlertController.m:225]: alertView will only ever be a casted self.view.  It is used in place of self.view throughout.
I believe it makes sense to set here so there is never a time self.view != self.alertView.

adjusting... moved to viewDidLoad.

File: components/Dialogs/src/MDCAlertController.m (hunk 4 above)
Comment (ianegordon) [reply components/Dialogs/src/MDCAlertController.m:175]: Unfortunately that won't work if alertView hasn't been instantiated yet.

File: components/Dialogs/src/MDCAlertController.m (hunk 5 above)
Comment (ianegordon) [reply components/Dialogs/src/MDCAlertController.m:179]: Unfortunately that won't work if alertView hasn't been instantiated yet.

File: components/Dialogs/src/MDCAlertController.h (hunk 6 above)
Comment (ianegordon) [reply components/Dialogs/src/MDCAlertController.h]: Gone.

File: components/Dialogs/src/MDCAlertControllerView.h (hunk 7 above)
Comment (ianegordon) [reply components/Dialogs/src/MDCAlertControllerView.h]: Gone.

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m (hunk 9 above)
Comment (ianegordon) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m]: Gone.

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m (hunk 10 above)
Comment (ianegordon) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m]: Gone.

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m (hunk 11 above)
Comment (ianegordon) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m]: Gone.

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m (hunk 12 above)
Comment (ianegordon) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m]: We do not.  Removed from VC.

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m (hunk 13 above)
Comment (ianegordon) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m]: We do not.  Removed from VC.

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m (hunk 8 above)
Comment (ianegordon) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m]: Gone.

**Review by yarneo — APPROVED** [review]

**Pull request merged**